lib/src/value/calculation.dart

```diff
@@ -158,6 +382,197 @@ class SassCalculation extends Value {
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value mod(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    if (dividend is! SassNumber || modulus is! SassNumber) {
+      return SassCalculation._("mod", args);
+    }
+    _verifyCompatibleNumbers(args);
+    return dividend.modulo(modulus);
+  }
+
+  /// Creates a `round()` calculation with the given [strategyOrNumber], [numberOrStep], and [step].
+  /// Strategy must be either nearest, up, down or to-zero.
+  ///
+  /// Number and step must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value round(Object strategyOrNumber,
+      [Object? numberOrStep, Object? step]) {
+    strategyOrNumber = _simplify(strategyOrNumber);
+    numberOrStep = numberOrStep.andThen(_simplify);
+    step = step.andThen(_simplify);
+
+    var args = (strategyOrNumber, numberOrStep, step);
+
+    switch (args) {
+      case (SassNumber number, null, null):
+        return SassNumber(number.value.round().toDouble())
+            .coerceToMatch(number);
+
+      case (SassString strategy, SassNumber(), null)
+          when ({'nearest', 'up', 'down', 'to-zero'}.contains(strategy.text)):
+        throw SassScriptException("If strategy is not null, step is required.");
+
+      case (SassNumber number, SassNumber step, null)
+          when (!number.hasUnit('%') && !step.hasUnit('%')):
+        _verifyCompatibleNumbers([number, step]);
+        return number_lib.roundWithStep(SassString('nearest'), number, step);
+
+      case (SassString strategy, SassNumber number, SassNumber step)
+          when (!strategy.isVar && !number.hasUnit('%') && !step.hasUnit('%')):
+        _verifyCompatibleNumbers([number, step]);
+        return number_lib.roundWithStep(strategy, number, step);
+
+      case (Object strategy, Object(), Object())
+          when (strategy is! SassString ||
+              !{'nearest', 'up', 'down', 'to-zero'}.contains(strategy.text) &&
+                  !strategy.isVar):
```

Comment:
It's generally easier to understand a `switch` if you add more `case` clauses when possible rather than complex `when` conditions:

```dart
case (SassString(text: 'nearest' || 'up' || 'down' || 'to-zero'), _?, _?):
case (SassString(isVar: true), _?, _?):
  // Fall through to create an unsimplified calculation

case (_, _?, _?):
  throw // ...
```

Reply:
Done!